analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd

```diff
@@ -0,0 +1,227 @@
+
+- LGG, IDH
+harbors an IDH R132 mutation
+
+ - LGG, H3.3
+harbors an H3F3A K28M or G35R/V mutation
+
+ -  LGG, H3.1
+harbors an HIST1H3B K28M
+harbors and HIST1H3C  K28M
+
+
+### Setup
+```{r}
+library(tidyverse)
+
+# Look for git root folder
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# get subset folder
+subset_dir <- file.path(root_dir, 
+                        "analyses",
+                        "molecular-subtyping-LGAT",
+                        "lgat-subset")
+
+# create if doesn't exist
+if (!dir.exists(subset_dir)) {
+  dir.create(subset_dir)
+}
+```
+
+### Input
+
+```{r}
+# File from 00-LGAT-select-pathology-dx that is used for the pathology diagnosis
+# inclusion/exclusion criteria
+path_dx_list <- jsonlite::fromJSON(
+  file.path(subset_dir, 
+            "lgat_subtyping_path_dx_strings.json")
+)
+
+# clinical file
+clinical <- read_tsv(file.path(root_dir, 
+                              "data",
+                              "pbta-histologies.tsv"), 
+                    guess_max = 10000)
+
+# consensus mutation data
+consensusMutation <- read_tsv(file.path(root_dir,
+                                        "data",
+                                        "pbta-snv-consensus-mutation.maf.tsv.gz"))
+
+# combined list for SNV of interest per subtype
+# additional information for genes with known hotspots/canonical mutation 
+# is also provided
+snvOI <- jsonlite::fromJSON(file.path(root_dir,
+                                      "analyses",
+                                      "molecular-subtyping-LGAT",
+                                      "input",
+                                      "snvOI_list.json"))
+
+```
+
+Collase multiple hotspots in genes with "|" so easy grep calls
+
+```{r}
+BRAF_hotspot <-paste(snvOI$BRAF_V600E$hotspot[!is.na( snvOI$BRAF_V600E$hotspot)],collapse = "|")
+FGFR_hotspot <-paste(snvOI$FGFR$hotspot[!is.na( snvOI$FGFR$hotspot)],collapse = "|")
+IDH_hotspot<- paste(snvOI$IDH$hotspot[!is.na( snvOI$IDH$hotspot)],collapse = "|")
+```
+
+Filter to tumor samples that should be included on the basis of pathology diagnosis
+```{r}
+lgat_specimens_df <- clinical %>%
+  dplyr::filter(str_detect(str_to_lower(pathology_diagnosis),  # Inclusion criteria
+                    paste0(path_dx_list$include_path_dx, collapse = "|")),
+         # Exclusion criteria
+         str_detect(str_to_lower(pathology_diagnosis),
+                    paste0(path_dx_list$exclude_path_dx, collapse = "|"),
+                    negate = TRUE),
+         # Tumors
+         sample_type == "Tumor",
+         composition == "Solid Tissue")
+
+# Write this intermediate file to the subset directory as it allows for
+# inspection
+write_tsv(lgat_specimens_df, file.path(subset_dir, "lgat_metadata.tsv"))
+
+# Filter to WGS samples
+lgat_wgs_df <- lgat_specimens_df %>%
+  dplyr::filter(experimental_strategy == "WGS") %>%
+  dplyr::select(Kids_First_Biospecimen_ID)
```

Comment:
If the purpose here is to filter out RNA-Seq samples, then I would do an exclusion instead, which would allow flexibility in the rare case WXS are added later.

Reply:
I implemented the filter to only remove RNA-Seq , but I've kept this as dataframe added sample_id since we would need the sample_id to match with RNA-Seq biospecimens in the final step.

Reply:
good call